include/authselect.h

```diff
@@ -400,6 +411,79 @@ authselect_path_dconf_db();
 const char *
 authselect_path_dconf_lock();
 
+/**
+ * Empty value for initialization.
+ */
+#define AUTHSELECT_SYMLINK_NONE     0x0000
+
+/**
+ * Create symbolic links for meta files.
+ */
+#define AUTHSELECT_SYMLINK_META     0x0001
+
+/**
+ * Create symbolic links for nsswitch.conf.
+ */
+#define AUTHSELECT_SYMLINK_NSSWITCH 0x0002
+
+/**
+ * Create symbolic links for PAM files.
+ */
+#define AUTHSELECT_SYMLINK_PAM      0x0004
+
+/**
+ * Create symbolic links for dconf files.
+ */
+#define AUTHSELECT_SYMLINK_DCONF    0x0008
+
+/**
+ * Create new profile.
+ *
+ * Create new profle named @name. Argument @type determines the type of
+ * newly created profile and can be either AUTHSELECT_PROFILE_VENDOR or
+ * AUTHSELECT_PROFILE_CUSTOM. Default profiles cannot be created as they
+ * are only shipped with installation.
+ *
+ * New profile can be based on an existing profile @base_id. Argument
+ * @base_type determines where the base profile is located. If it is
+ * AUTHSELECT_PROFILE_ANY, standard algorithm is used, that means if
+ * custom/ prefix is is @base_id it is a custom profile, otherwise
+ * it is search in vendor and then in default profiles.
+ *
+ * When the profile is based on @base_id, files from @base_id are copied
+ * in the new profile location. Symbolic links can be created instead of
+ * creating a copy of a file by providing a bit mask consisting of:
+ * - AUTHSELECT_SYMLINK_META
+ * - AUTHSELECT_SYMLINK_NSSWITCH
+ * - AUTHSELECT_SYMLINK_PAM
+ * - AUTHSELECT_SYMLINK_DCONF
+ *
+ * Alternatively, specific files that should be symlinked can be specified
```

Comment:
What is the use-case for this?

Reply:
For example, you can call `authselect create-profile my-profile -b sssd --symlink-pam` and only  pam stacks will be symlinked, other files will be copied and you can edit them. So you don't have to maintain the pam stack, but you can edit e.g. nsswitch.conf.

If you do not want to symlink the all pam files, but e.g. only postlogin, you can run `authselect create-profile my-profile -b sssd --symlink postlogin`